silq/parameters/acquisition_parameters.py

```diff
@@ -1965,6 +1965,69 @@ def analyse(self, traces: Dict[str, Dict[str, np.ndarray]] = None):
         return results
 
 
+class EDSRParameter(NMRParameter):
+    """
+    Parameter for EDSR measurements based on NMR parameter and pulse sequence
+
+    Refer to NMRParameter for details. In addition to all properties copied from NMRParameter,
+    EDSRParameter includes analysis of electron readout right after EDSR(NMR) pulse during
+    NMR['post_pulse'] = DCPulse('read') that needs to be present in NMRPulseSequence.
+
+    Args:
+        Refer to NMRParameter
+    Parameters:
+        Refer to NMRParameter
+    """
+
+    def __init__(self, name: str = 'EDSR',
+                 names: List[str] = ['flips', 'flip_probability', 'up_proportions',
+                                     'state_probability', 'threshold_up_proportion',
+                                     'EDSR_up_proportion'],
+                 **kwargs):
+        super().__init__(name=name,
+                         names=names,
+                         **kwargs)
+
+    @property
+    def names(self):
+        names = super().names
+        names.append('EDSR_up_proportion')
+        return names
+
+    @names.setter
+    def names(self, names):
+        self._names = names
+
+    def analyse(self, traces: Dict[str, Dict[str, np.ndarray]] = None):
+        """
+        Reading out electron spin-up proportion after EDSR (NMR) pulse during 'read' DCPulse.
+
+        Returns:
+            (Dict[str, Any]): Dict containing:
+            * all results from NMRParameter
+            * **EDSR_up_proportion**: electron spin-up proportion right after EDSR (NMR) pulse.
+        """
+        results = super().analyse(traces)
+
+        # Extract points for read DC pulse after EDSR pulse
+        EDSR_read_traces_name = f"{self.NMR['post_pulse'].name}"
```

Comment:
This does not need to be converted to a string, shouldn't it already be a string? If not, then simply use str(self.NMR['post_pulse'].name)

Reply:
Also, how is `post_pulse` defined? There's no pulse added to the class, so won't this by default raise an error?

Reply:
And isn't this variable used just once? I don't see the reason of creating a separate variable for this

Reply:
post_pulse is defined by default in the NMRPulseSequence to be DCPulse('read', acquire=True). So there shouldn't be errors.
Otherwise, I've already corrected your suggestions.